solution-base/operator.yaml

```
@@ -0,0 +1,390 @@
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kubedb
+  template:
+    metadata:
+      labels:
+        app: kubedb
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      serviceAccountName: ${KUBEDB_SERVICE_ACCOUNT}
+      imagePullSecrets: [${KUBEDB_IMAGE_PULL_SECRET}]
+      containers:
+      - name: operator
+        image: ${KUBEDB_DOCKER_REGISTRY}/${KUBEDB_OPERATOR_NAME}:${KUBEDB_OPERATOR_TAG}
+        imagePullPolicy: ${KUBEDB_IMAGE_PULL_POLICY}
+        args:
+        - run
+        - --v=3
+        - --governing-service=kubedb
+        - --rbac=${KUBEDB_ENABLE_RBAC}
+        - --secure-port=8443
+        - --audit-log-path=-
+        - --tls-cert-file=/var/serving-cert/tls.crt
+        - --tls-private-key-file=/var/serving-cert/tls.key
+        - --enable-mutating-webhook=${KUBEDB_ENABLE_MUTATING_WEBHOOK}
+        - --enable-validating-webhook=${KUBEDB_ENABLE_VALIDATING_WEBHOOK}
+        - --enable-status-subresource=${KUBEDB_ENABLE_STATUS_SUBRESOURCE}
+        - --bypass-validating-webhook-xray=${KUBEDB_BYPASS_VALIDATING_WEBHOOK_XRAY}
+        - --use-kubeapiserver-fqdn-for-aks=${KUBEDB_USE_KUBEAPISERVER_FQDN_FOR_AKS}
+        - --enable-analytics=${KUBEDB_ENABLE_ANALYTICS}
+        env:
+        - name: MY_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: MY_POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        ports:
+        - containerPort: 8443
+        volumeMounts:
+        - mountPath: /var/serving-cert
+          name: serving-cert
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: 8443
+            scheme: HTTPS
+          initialDelaySeconds: 5
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8443
+            scheme: HTTPS
+          initialDelaySeconds: 15
+          timeoutSeconds: 15
+        resources:
+          requests:
+            cpu: "100m"
+      nodeSelector:
+        beta.kubernetes.io/os: linux
+        beta.kubernetes.io/arch: amd64
+      volumes:
+      - name: serving-cert
+        secret:
+          defaultMode: 420
+          secretName: kubedb-${KUBEDB_OPERATOR_NAME}-apiserver-cert
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Issuer
+metadata:
+  name: selfsigned-issuer
+  namespace: ${KUBEDB_NAMESPACE}
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: kubedb-operator-apiserver-cert
+  namespace: ${KUBEDB_NAMESPACE}
+spec:
+  # Secret names are always required.
+  secretName: kubedb-operator-apiserver-cert
+  # TODO: How does kubedb handle cert regens?
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  isCA: false
+  keyAlgorithm: rsa
+  keyEncoding: pkcs1
+  keySize: 2048
+  # At least one of a DNS Name, URI, Email Address, or IP address is required.
+  dnsNames:
+  - kubedb-operator
```

Adding this here before I forget: should be modified to 
```suggestion
  - kubedb-${KUBEDB_OPERATOR_NAME}
```
Because it corresponds to the service name, and that's how we templated it's value on line 117